lib/rules/no-useless-constructor.js

```diff
@@ -177,7 +180,15 @@ module.exports = {
             if (superClass ? isRedundantSuperCall(body, ctorParams) : (body.length === 0)) {
                 context.report({
                     node,
-                    messageId: "noUselessConstructor"
+                    messageId: "noUselessConstructor",
+                    suggest: [
+                        {
+                            messageId: "removeConstructor",
+                            fix(fixer) {
+                                return fixer.remove(node);
```

Comment:
In some edge cases, if the constructor triggers automatic semicolon insertion, then the semicolon needs to be inserted explicitly for the code to continue working when the constructor is removed. For example:

```js
class A {
  foo = "bar"
  constructor() { }
  [0]() { }
}
```

Simply removing the constructor from the above code would result in a syntax error. In this case, the fix needs to insert the semicolon:

```js
class A {
  foo = "bar"
  ;
  [0]() { }
}
```

The logic to conditionally insert a semicolon while replacing a node is already implemented in other rules, for example in `require-await`:

https://github.com/eslint/eslint/blob/3a4eaf921543b1cd5d1df4ea9dec02fab396af2a/lib/rules/require-await.js#L111-L114

For our specific case, calculating `addSemicolon` should be a bit easier because we can be sure that the node to be removed remove is a constructor, so probably:

```js
const nextToken = sourceCode.getTokenAfter(node);
const addSemiColon = nextToken.type === "Punctuator" && nextToken.value === "[" && astUtils.needsPrecedingSemicolon(sourceCode, node);                
return fixer.replaceTextRange(node.range, addSemiColon ? ";" : "");
```

Reply:
Hey, @fasttime! I've made some changes that I believe address your concerns. Let me know what you think :)

Reply:
Thanks for the update @imjordanxd. Unfortunately, this is still not working as expected for the above edge case. I left some comments in code to clarify. Just let me know if something is not clear.